src/ripple/protocol/impl/RippleAddress.cpp

```diff
@@ -813,4 +845,62 @@ RippleAddress RippleAddress::createSeedGeneric (std::string const& strText)
     return naNew;
 }
 
+uint256 keyFromSeed (uint128 const& seed)
+{
+    Serializer s;
+
+    s.add128 (seed);
+    uint256 result = s.getSHA512Half();
+
+    s.secureErase ();
+
+    return result;
+}
+
+RippleAddress getSeedFromRPC (Json::Value const& params)
+{
+    bool const has_passphrase = params.isMember (jss::passphrase);
+    bool const has_seed       = params.isMember (jss::seed);
+    bool const has_seed_hex   = params.isMember (jss::seed_hex);
+
+    int const n_secrets = has_passphrase + has_seed + has_seed_hex;
+
+    if (n_secrets > 1)
+    {
+        // `passphrase`, `seed`, and `seed_hex` are mutually exclusive.
+        return RippleAddress();
+    }
+
+    RippleAddress result;
+
+    if (has_seed)
+    {
+        std::string const seed = params[jss::seed].asString();
+
+        result.setSeed (seed);
+    }
+    else if (has_seed_hex)
+    {
+        uint128 seed;
+        std::string const seed_hex = params[jss::seed_hex].asString();
+
+        if (seed_hex.size() != 32  ||  !seed.SetHex (seed_hex, true))
+        {
+            return RippleAddress();
+        }
+
+        result.setSeed (seed);
+    }
+    else if (has_passphrase)
+    {
+        std::string const passphrase = params[jss::passphrase].asString();
+
+        // Given `key_type`, `passphrase` is always the passphrase.
```

Comment:
This comment doesn't make any sense. `key_type` is not referenced in this function.